keps/sig-auth/20190116-service-account-external-signing.md

```diff
@@ -119,18 +172,90 @@ message ListPublicKeysResponse {
 
 ### Implementation Details/Notes/Constraints
 
-The API server flag `--service-account-key-file` can be specified multiple times for legacy SA tokens and projected tokens. Validation keys from this flag will be merged with the response of `ListPublicKeys()`. A new flag `--key-service-url` will be added to the API server specifying a unix socket where the key service will be accessible.
+The API server flag `--service-account-key-file` can be specified multiple times
+for legacy SA tokens and projected tokens. Validation keys from this flag will
+be merged with the response of `ListPublicKeys()`. A new flag
+`--key-service-url` will be added to the API server specifying a unix socket
+where the key service will be accessible.
 
 ### Risks and Mitigations
 
-New token generation and validation could suffer a performance difference when reading over a socket, as an external process will be signing data.
+New token generation and validation could suffer a performance difference when
+reading over a socket, as an external process will be signing data or delegating
+off host.
+
+Signing and verifying tokens over a grpc API carries the risk of a server side
+request forgery, where a malicious client could generate tokens. To mitigate
+this risk, the API will only be accessible over a unix socket.
+
+Additionally, if the signing server becomes unavailable, creation of new tokens
+and verificaiton of existing tokens would be impacted.
+
+## Design Details
+
+### Test Plan
+
+Unit tests covering:
+* ProjectedToken signing works with external signer client
+* OIDC keys URL shows public keys from external signer
 
-Signing and verifying tokens over a grpc API carries the risk of a server side request forgery, where a malicious client could generate tokens. To mitigate this risk, the API will only be accessible over a unix socket.
+Integration tests covering:
+* ProjectedToken signing works with simple signer server
+* OIDC keys URL shows public keys from external signer
+* Validate apiserver flag logic (require issuer to be set)
+* Validate that legacy Service Account tokens can be validated while external
+    signer is down
```

Comment:
Legacy and new style tokens should both be validate-able while the external signer is down at least for some period of time because we should cache the public keys.